Maven/Tycho build

hefloryd opened the pull request.

hefloryd committed:
add capra feature project

hefloryd committed:
add capra update site project

hefloryd committed:
fix plugin versions

hefloryd committed:
add target platform

hefloryd committed:
rename test CreateTrace to TestCreateTrace for automatic discovery of tests when building headless

hefloryd committed:
poms for Maven/Tycho build

hefloryd committed:
add missing plugin .gitignore

hefloryd committed:
add generated code

Code generation does not work when building headless. See
https://bugs.eclipse.org/bugs/show_bug.cgi?id=487683

hefloryd committed:
update README.md

steghoja commented on README.md:
```diff
@@ -14,19 +14,15 @@ Capra uses the Eclipse Modelling Framework (EMF) as its base technology and stor
 * Open your Eclipse Environment
 * Go to File >> Import and select Git >> Projects from Git
 * Use the [GitHub repository](https://github.com/Salome-Maro/TraceabilityManagement) and import all available projects to your workspace
+* Set the target platform by opening org.eclipse.capra.releng.target/org.eclipse.capra.releng.target.target, and click on "Set as Target Platform" in the upper right corner. Eclipse will now download all dependencies. Note that this will take a while. Check the progress in the lower right corner.
```
I wonder if it is worthwhile to list the dependencies somewhere. There is always a chance something goes wrong with the automatic download processes and having a list of stuff that is required (and potentially the impact of not having it installed) can be helpful.

hefloryd replied:
Possibly, although you could argue that the .target-file is the list of dependencies. Having dependencies listed separately as well means updating information in two places whenever something changes.

steghoja replied:
True. However, especially to novices, these conventions might not be obvious. Maybe we can add something that says that this is where all dependencies can be found?

steghoja commented on org.eclipse.capra.ui.plantuml/.gitignore:
```diff
@@ -0,0 +1,2 @@
+/bin
+/target
```
I had added these directories to the main .gitignore and deleted this file. It worked just fine for me. Any particular reason to reintroduce it?

hefloryd replied:
Yes, I added it because Eclipse has previously created these automatically for me. It is also consistent with the other plugins in the project, and with what amalthea-src did, and I believe other Eclipse projects.

However it didn't create it for me now when I tried to repeat it. Perhaps it has to do with how you import the projects. 

If it isn't required I suggest we create a separate pull-request to remove all plugin .gitignores to keep it consistent.

steghoja replied:
Right, I did not realise that each plugin has its own .gitignore. While I feel like that is superfluous, I understand your argument to keep it consistent. If these files are automatically generated, we can just leave them in place.

hefloryd committed:
add .travis.yml

hefloryd commented:
I updated README,md with a note about dependencies. Later we can add an Oomph setup file which would automate this somewhat.

Also added .travis.yml for automated builds using Travis CI.

steghoja commented:
Thanks Hans-Erik!
